WcaOnRails/app/models/competition_event.rb

```diff
@@ -21,4 +21,31 @@ def to_wcif
       "rounds" => self.rounds.map(&:to_wcif),
     }
   end
+
+  def load_wcif!(wcif)
+    # Remove extra rounds.
+    self.rounds.each do |round|
+      wcif_round = wcif["rounds"][round.number - 1]
```

Comment:
If you're interested in the "static" event structure it can be awkward that it comes with 100x result information you don't care about. It also makes it harder to see if the events have changed. It's not so much about static or dynamic, more that often you only want one part and not the other.

I suspect our "merging" problems are mostly independent of this structure.

I can imagine an interface convention where you specify a "path" to a subtree in the WCIF, XPath style.

I'm just speculating here. I don't know of any change that is clearly better. This format will change many more times as it meets reality, and we'll learn as it happens.